### PR TITLE
WEBUI-1414: update functional test for validation 3.0.x

### PR DESCRIPTION
**Pull request by Nishant0928**
https://jira.nuxeo.com/browse/WEBUI-1414

### CODE REVIEW

File: packages/nuxeo-web-ui-ftest/features/step_definitions/document.js [hunk 1]
```diff
@@ -338,22 +338,39 @@ Then(/^I can unpublish the document$/, async function() {
...
-  const form = browser.editForm(docType);
-  form.waitForVisible();
-  driver.waitUntil(
-    () => form.errorMessages.length === nbErrors,
-    `Expecting to get ${nbErrors} results but found ${form.errorMessages.length}`,
+Then('I can see {int} validation error(s) in the {string} edit form', async function(nbErrors, docType) {
+  const browser = await this.ui.browser;
+  const form = await browser.editForm(docType);
+  await form.waitForVisible();
+
```
Comment (rahuljain-dev) [inline packages/nuxeo-web-ui-ftest/features/step_definitions/document.js]: remove extra line

File: packages/nuxeo-web-ui-ftest/features/step_definitions/document.js [hunk 2]
```diff
@@ -338,22 +338,39 @@ Then(/^I can unpublish the document$/, async function() {
...
+  await driver.waitUntil(
+    async () => {
+      await driver.pause(1000); // Add a small delay if needed
+      const errorMessages = await form.errorMessages;
+      return errorMessages.length === nbErrors;
+    },
+    {
+      timeout: 10000, // Adjust the timeout as needed
```
Comment (rahuljain-dev) [inline packages/nuxeo-web-ui-ftest/features/step_definitions/document.js]: we don't need this much timeout please look for alternative

File: packages/nuxeo-web-ui-ftest/features/step_definitions/document.js [hunk 3]
```diff
@@ -338,22 +338,39 @@ Then(/^I can unpublish the document$/, async function() {
...
+  const { browser } = await this.ui;
+
+  try {
+    const form = await browser.editForm(docType);
+    await form.waitForVisible();
+
+    const errorMessages = await form.errorMessages;
+
+    const hasErrorMessage = errorMessages.some((err) => err === message);
+
```
Comment (rahuljain-dev) [inline packages/nuxeo-web-ui-ftest/features/step_definitions/document.js]: remoce extra lines

File: packages/nuxeo-web-ui-ftest/pages/ui/browser/document_form_layout.js [hunk 4]
```diff
@@ -17,7 +18,18 @@ export default class DocumentFormLayout extends BasePage {
...
+    return (async () => {
+      const errorElements = await this.el.$$('#error .error');
+      const errorTexts = [];
+
+      for (let i = 0; i < errorElements.length; i++) {
+        const errorElement = errorElements[i];
+        const errorText = await errorElement.getText();
+        errorTexts.push(errorText);
+      }
+
```
Comment (rahuljain-dev) [inline packages/nuxeo-web-ui-ftest/pages/ui/browser/document_form_layout.js]: remove extra lines

**Review by alokhyland — APPROVED** [review]

**Review by MansaQA — APPROVED** [review]

**Review by rahuljain-dev — APPROVED** [review]

**Review by arupmmi07 — APPROVED** [review]

**Pull request merged**